src/concrete/ml/sklearn/tree_to_numpy.py

```diff
@@ -333,6 +351,129 @@ def tree_to_numpy(
     # but also rounding the threshold such that they are now integers
     q_y = tree_values_preprocessing(onnx_model, framework, output_n_bits)
 
-    _tree_inference, onnx_model = get_equivalent_numpy_forward_from_onnx(onnx_model)
+    _tree_inference, onnx_model = get_equivalent_numpy_forward_from_onnx(
+        onnx_model, check_model=True, lsbs_to_remove=lsbs_to_remove
+    )
 
     return (_tree_inference, [q_y.quantizer], onnx_model)
+
+
+# Remove this function once the truncate feature is released
+# FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/4143
+def _compute_lsb_to_remove_for_trees(
```

Comment:
The complexity in this function is real. We basically reproduce the tree inference by hand to compute the lsbs_to_remove. If we really need to do this, we might want to use our `get_equivalent_numpy_forward_from_onnx` to get intermediate values so that we don't have to reimplement everything from scratch.

If this is removed with the truncate feature then I would advise we go for truncate as soon as available removing the complexity of this function.